platform/command/new_command.go

```diff
@@ -69,3 +92,65 @@ func MakeNewCommandEndpoint(svc Service) endpoint.Endpoint {
 		return newCommandResponse{Payload: payload}, nil
 	}
 }
+
+type RawCommand struct {
+	UDID        string `json:"udid" plist:"-"`
+	CommandUUID string `json:"command_uuid"`
+	Command     struct {
+		RequestType string `json:"request_type"`
+	} `json:"command"`
+	Raw []byte `plist:"-" json:"payload"`
+}
+
+type newRawCommandRequest struct {
+	RawCommand
+}
+
+type newRawCommandResponse struct {
+	Payload *RawCommand `json:"payload,omitempty"`
+	Err     error       `json:"error,omitempty"`
+}
+
+func (r newRawCommandResponse) Failed() error   { return r.Err }
+func (r newRawCommandResponse) StatusCode() int { return http.StatusCreated }
+
+func decodeNewRawCommandRequest(ctx context.Context, r *http.Request) (interface{}, error) {
+	udid, ok := mux.Vars(r)["udid"]
+	if !ok {
+		return nil, errors.New("empty udid")
+	}
+
+	payload, err := io.ReadAll(r.Body)
+	if err != nil {
+		return nil, fmt.Errorf("read payload body: %w", err)
+	}
+
+	// verify body is valid plist and parse CommandUUID and RequestType
+	var req newRawCommandRequest
+	if err := plist.NewXMLDecoder(bytes.NewBuffer(payload)).Decode(&req); err != nil {
+		return nil, fmt.Errorf("parse payload as plist: %w", err)
+	}
+
+	req.UDID = udid
+	req.Raw = payload
+	return req, nil
+}
+
+var errMalformedRequest = errors.New("request is malformed")
+
+// MakeNewRawCommandEndpoint creates an endpoint which creates new raw MDM Commands.
+func MakeNewRawCommandEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(newRawCommandRequest)
+		if req.UDID == "" {
+			return newRawCommandResponse{Err: errEmptyRequest}, nil
+		}
+		if req.CommandUUID == "" || req.Command.RequestType == "" {
```

Comment:
maybe also check for an empty/nil payload?

Reply:
The request (type `newRawCommandRequest`) is parsed by `decodeNewRawCommandRequest` before being handed to the Endpoint. This parses the body as a plist, so it guarantees that the payload is non-nil and contains raw plist, though not necessarily a properly formatted command.

Reply:
The decoder could also guarantee the command uuid and request tyke are also set tho, yeah? Negating the need for those checks?

Reply:
For sure. Again, it mostly goes back to matching existing code (e.g. the non-raw command endpoints) which: